src/main/java/org/jitsi/jicofo/bridge/BridgeSelectionStrategy.java

```diff
@@ -67,79 +69,120 @@ public Bridge select(
+     * bitrate that they're handling, which is used an indirect load indicator).
+     * @param conferenceBridges the bridges that are in the conference.
      * @param participantRegion the region of the participant for which a
      * bridge is to be selected.
      * @return the selected bridge, or {@code null} if no bridge is
      * available.
      */
-    private Bridge selectInitial(List<Bridge> bridges,
+    public Bridge doSelect(List<Bridge> bridges,
+                                 List<Bridge> conferenceBridges,
                                  String participantRegion)
     {
-        Bridge bridge = null;
-
-        // Prefer a bridge in the participant's region.
-        if (participantRegion != null)
+        if (bridges.isEmpty())
         {
-            bridge = findFirstOperationalInRegion(bridges, participantRegion);
+            return null;
         }
 
-        // Otherwise, prefer a bridge in the local region.
-        if (bridge == null)
-        {
-            bridge = findFirstOperationalInRegion(bridges, localRegion);
-        }
+        double maxStress = findStressLimit(bridges);
 
-        // Otherwise, just find the first operational bridge.
-        if (bridge == null)
-        {
-            bridge = findFirstOperationalInRegion(bridges, null);
-        }
+        // Try the first operational bridge in the participant region and
+        // that is already in the conference.
+        return bridges.stream()
+            .filter(stressIsLessThanOrEqual(maxStress))
+            .filter(selectFrom(conferenceBridges))
+            .filter(inRegion(participantRegion))
+            .findFirst()
+            // Otherwise, try to add a new bridge in the participant region.
+            .orElse(bridges.stream()
+                .filter(stressIsLessThanOrEqual(maxStress))
+                .filter(inRegion(participantRegion))
+                .findFirst()
+                // Otherwise, try to add a new bridge that is already in the
+                // conference.
+                .orElse(bridges.stream()
+                    .filter(stressIsLessThanOrEqual(maxStress))
+                    .filter(selectFrom(conferenceBridges))
+                    .findFirst()
+                    // Otherwise, try to add a new bridge in the local
+                    // region.
+                    .orElse(bridges.stream()
+                        .filter(stressIsLessThanOrEqual(maxStress))
+                        .filter(inRegion(localRegion))
+                        .findFirst()
+                        // Otherwise, try to add the least loaded bridge
+                        // that we know of.
+                        // TODO: perhaps use a bridge in a nearby region (if
+                        // we have data about the topology of the regions).
+                        .orElse(bridges.stream()
+                            .filter(stressIsLessThanOrEqual(maxStress))
```

Comment:
This never considers bridges with `stress > maxStress`, failing instead. At least as a start we should preserve the current behavior and fallback to a stressed bridge instead of failing.

Reply:
this part has changed and we're now considering all bridges, even those that are very highly stressed